app/controllers/admin/homes_controller.rb

```diff
@@ -7,4 +7,20 @@ def index
       render file: "#{Rails.root}/public/404.html", status: 404
     end
   end
+
+  def suspend
+    home_to_suspend = Home.find(params[:id])
+    home_to_suspend.suspend if current_admin?
```

Comment:
checking for current admin should be done by the base admin controller. That before action wasnt triggering due to 
class Admin::HomesController < ApplicationController
should be
class Admin::HomesController < Admin::BaseController
The admin base controller inherits from ApplicationController bringing in that functionality as well.